enaml/qt/qt_mpl_canvas.py

```diff
@@ -89,11 +89,17 @@ def _refresh_mpl_widget(self):
         # visible, or QVBoxLayout will ignore it for size hinting.
         figure = self.declaration.figure
         if figure:
-            canvas = FigureCanvasQTAgg(figure)
+            if isinstance(figure.canvas, FigureCanvasQTAgg):
+                canvas = figure.canvas
+                canvas.manager.toolbar = None
+                toolbar = canvas.toolbar
+                toolbar.setParent(widget)
```

Comment:
This call to `setParent` is not needed. The layout will automatically reparent the widget when it's added to the layout.